tests/test_builder.py

```diff
@@ -0,0 +1,424 @@
+import math
```

Comment:
Generic: How do I run all the tests. Why not add pytest or so to run all the tests as a check (also in the pipeline)

Reply:
The standard way is to use pytest. E.g.`pytest tests/test_builder.py` for a specific test file or `make tests` or `make external-tests` (see the `Makefile`) to run all tests. These tests are also already in the pipeline.